processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go

```diff
@@ -14,13 +14,94 @@
 
 package ecs
 
-import "net/http"
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
 
-type HTTPClient interface {
-	Do(req *http.Request) (*http.Response, error)
+type metadataProvider interface {
+	fetchTask(tmde string) (*TaskMetaData, error)
+	fetchContainer(tmde string) (*Container, error)
 }
 
-type ecsMetadataProvider interface {
-	fetchTaskMetaData(tmde string) (*TaskMetaData, error)
-	fetchContainerMetaData(tmde string) (*Container, error)
+type TaskMetaData struct {
```

Comment:
s/TaskMetaData/TaskMetadata? For consistency with other places where you use "Metadata" as one word?

Reply:
Done.